src/io_object.cpp

```diff
@@ -72,71 +72,206 @@ namespace ioutil {
         return ::stat(io->as_string()->c_str(), sb);
     }
 
-    flags_struct::flags_struct(Env *env, Value flags_obj) {
-        if (!flags_obj || flags_obj->is_nil())
-            return;
-
-        if (!flags_obj->is_integer() && !flags_obj->is_string()) {
-            if (flags_obj->respond_to(env, "to_str"_s)) {
-                flags_obj = flags_obj->to_str(env);
-            } else if (flags_obj->respond_to(env, "to_int"_s)) {
-                flags_obj = flags_obj->to_int(env);
+    namespace {
```

Comment:
Would it match Natalie style a bit more to have a Flags `class` with member functions vs this `namespace` and functions that accept `flags_struct`? I believe this is the only place we use this sort of style...

Reply:
I split that into #1247, right now I just kind of want to see the results.
This change is a pretty isolated one, the calling code will not see the results.

And for what it's worth: I chose this approach to reduce the compilation time: every private method in C++ means a change in the header file, which means we have to recompile pretty much everything. Since I had no clear vision of where I was going with this change (as can be seen by the 3 rewrites and countless related cases that prepared a thing or two for this change), this did help a bit.

Reply:
Great, works for me!